ci/docker/runtime_functions.sh

```diff
@@ -33,35 +33,23 @@ clean_repo() {
     git submodule update --init --recursive
 }
 
-build_ccache_wrappers() {
-    set -ex
-
-    rm -f cc
-    rm -f cxx
-
-    touch cc
-    touch cxx
-
-    if [ -z ${CC+x} ]; then
-        echo "No \$CC set, defaulting to gcc";
-        export CC=gcc
-    fi
-
-    if [ -z ${CXX+x} ]; then
-       echo "No \$CXX set, defaulting to g++";
-       export CXX=g++
-    fi
-
-    # this function is nessesary for cuda enabled make based builds, since nvcc needs just an executable for -ccbin
 
-    echo -e "#!/bin/sh\n/usr/local/bin/ccache ${CC} \"\$@\"\n" >> cc
-    echo -e "#!/bin/sh\n/usr/local/bin/ccache ${CXX} \"\$@\"\n" >> cxx
+install_ccache_wrappers() {
+    set -ex
 
-    chmod +x cc
-    chmod +x cxx
+    # Add to the beginning of path to ensure this redirection is picked up instead
```

Comment:
Why is this better than the previous solution? It looks less explicit and hacky. Can you provide some rationale?

Reply:
It is the recommended way by CCache: https://ccache.samba.org/manual.html#_run_modes
This way ensures that ccache is always used if the symlinks are set. Otherwise, we'd have to make a possibility in all build-paths (including third party) that they use ccache properly. This caused problems in compiling ps-lite, because that project had **another** third party dependency (ZeroMQ) which did not pick up the propagated compiler.